generators/init/config.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2013-2021 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const { version: jhipsterVersion } = require('../../package.json');
+
+/** Required config to be skipped */
+const requiredConfig = {
+  jhipsterVersion,
+  projectName: 'JHipster project',
```

Comment:
maybe missing prettierDefaultIndent and prettierJavaIndent, as the values must not be empty ?

Reply:
Not needed, it should only write the default value into .yo-rc.json.
I’ve forgot to update required config in generator-defaults.

Reply:
but I think without these values here, we have an issue:

```
➜ jhipster init --skip-prompts
INFO! Using JHipster version installed globally
⬢ Welcome to the JHipster Init ⬢
   create .prettierrc
   create .prettierignore
   create .gitignore
   create .gitattributes
   create .editorconfig
   create .huskyrc
   create package.json
   create .lintstagedrc.js
    force .yo-rc.json
   create README.md
Git repository initialized.

Changes to package.json were detected.
Error detecting the package manager. Falling back to npm.

Running npm install for you to install the required dependencies.

added 152 packages, and audited 153 packages in 3s

30 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Application commit to Git failed from /home/pgrimaud/tmp/01-beer. Try to commit manually.
Congratulations, JHipster execution is complete!
Sponsored with ❤️  by @oktadev.

18:15:13 in ~/tmp/01-beer on  master [+] took 5s 
➜ cat .yo-rc.json
{
  "generator-jhipster": {
    "baseName": "jhipster",
    "jhipsterVersion": "7.1.0",
    "projectName": "JHipster project"
  }
}
```

No git commit, because the prettierrc is incorrect:

```
# Prettier configuration

printWidth: 140
singleQuote: true
tabWidth: 
useTabs: false

# js and ts rules:
arrowParens: avoid

# jsx and tsx rules:
jsxBracketSameLine: false

# java rules:
overrides:
  - files: "*.java"
    options:
      tabWidth: 

```

Reply:
I have a fix, will push when I get a chance.

Reply:
should be fixed now.